nwb_conversion_tools/nwbconverter.py

```diff
@@ -96,18 +96,26 @@ def get_metadata(self):
             metadata = dict_deep_update(metadata, interface_metadata)
         return metadata
 
-    def run_conversion(self, metadata: dict, save_to_file: bool = True, nwbfile_path: Optional[str] = None,
-                       overwrite: bool = False, nwbfile: Optional[NWBFile] = None,
-                       conversion_options: Optional[dict] = None):
+    def run_conversion(
+            self, 
+            metadata: dict, 
+            save_to_file: Optional[bool] = True, 
+            nwbfile_path: Optional[str] = None,
+            overwrite: Optional[bool] = False, 
+            nwbfile: Optional[NWBFile] = None,
+            conversion_options: Optional[dict] = None
+        ):
         """Build nwbfile object, auto-populate with minimal values if missing."""
+
         assert (not save_to_file and nwbfile_path is None) or nwbfile is None, \
             "Either pass a nwbfile_path location with save_to_file=True, or a nwbfile object, but not both!"
+
         if conversion_options is None:
             conversion_options = dict()
         else:
             validate(instance=conversion_options, schema=self.get_conversion_options_schema())
 
-        if nwbfile is None:
+        if nwbfile is None and not Path(nwbfile_path).is_file():
```

Comment:
This will cause an error when `nwbfile_path` defaults, since Path(None) expects `str `not `NoneType`.

But I see we do not currently support the use case of reading an existing NWBFile, and then returning the object post conversion (i.e., nwbfile=None, nwbfile_path="my_nwbfile.nwb", save_to_file=False).

Also we do not currently support passing an NWBFile object, then writing it post conversion (i.e., nwbfile=my_nwbfile_object, save_to_file=True, nwbfile_path="my_nwbfile.nwb").

While Ben has made it clear that adding support for such things can stretch us thin since each use-case represents an additional mode we have to provide support for with increasing complexity along each dimension, we should at a bare minimum improve the docstring for this function to make it clear how it is intended to be used. Granted if a user is simply using it as intended in the standard template pipeline, with most of the arguments being defaulted, there shouldn't be any issue.